sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/SubResource.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// A class representing a sub-resource that contains only the ID.
+    /// </summary>
+    [ReferenceType]
+    public abstract class SubResource
+    {
+        /// <summary>
+        /// ARM resource identifier.
+        /// </summary>
+        /// <value></value>
+        public virtual ResourceIdentifier Id { get; set; }
```

Comment:
This should be get only with a protected constructor that takes in the id and sets it.

This is for deserialization scenarios

Reply:
I put a todo in this file for us to consider how to make `SubResource`/`SubResourceReadOnly` to follow the current pattern (in `Resource` and `TrackedResource`) that the read-only one should be the default.
And I added serilization code. Thanks for the comments!